### DISCUSSION TITLE
Better visual information with a led bar

### DISCUSSION THREAD

**Discussion by jdgarcia99** [OpenEVSE/openevse_esp32_firmware#891] [Show and tell]
I missed some visual effects other market chargers have with regards what the EVSE is doing, so I decided to modify mine to improve a little bit the user experience.

IMHO led bars are powerfull as they can show you the EVSE status in a nutshell, this is EVSE status, errors and, in particular, EVSE activity when charging a car.

I decided to try with led strips and adding the WS2812FX library for led effects, so when my car is loading, the OpenEVSE shows a bar animation  that syncs with the current flow amps (the more amps, the more speed on the animation).

In case of any special error such as no with or network connection, the bar blinks to show special event to take action.

The color of the bar is synced with the OpenEVSE colors, so at this point this is pretty much familiar for any OpenEVSE user.

Also led bar bright is in sync with bright level from OpenEVSE so you can adjust bright intensity directly from regular web interface.

To do this, only had to modify the LedManagerTask source to introduce this new user interaction through the led bar.

Here is a video and some pictures of the OpenEVSE working with this modification.

https://github.com/user-attachments/assets/9548fec4-9b10-4377-ae12-56da410b780c

![IMG_2651](https://github.com/user-attachments/assets/50391d9d-2dcd-44c9-bacd-6c80bb85f7b4)
![IMG_2650](https://github.com/user-attachments/assets/dcb84136-a5da-4086-a372-7aa2f38df942)

In the event of more people interested I can Pull to the project

**Comment by jeremypoulter** [comment]
That is great, LED animations are something we are definitely missing, it would be great if you could raise a PR for this.

**Reply by jdgarcia99** [reply]
PR Created [https://github.com/OpenEVSE/openevse_esp32_firmware/pull/896](https://github.com/OpenEVSE/openevse_esp32_firmware/pull/896). Let me know if it is OK

**Comment by jdgarcia99** [comment]
Resynced with Global master. Now PR is [https://github.com/OpenEVSE/openevse_esp32_firmware/pull/899](https://github.com/OpenEVSE/openevse_esp32_firmware/pull/899))
